src/worldmap/worldmap_state.cpp

```diff
@@ -114,117 +115,101 @@ WorldMapState::load_state()
     // Create a new initial save.
     save_state();
   }
-  sq_settop(vm.get_vm(), oldtop);
 
   m_worldmap.m_in_level = false;
 }
 
 
 /** Load Tux **/
 void
-WorldMapState::load_tux()
+WorldMapState::load_tux(const ssq::Table& table)
 {
-  WORLDMAP_STATE_SQUIRREL_VM_GUARD;
-  WORLDMAP_STATE_SECTOR_GUARD;
+  WorldMapSector& sector = m_worldmap.get_sector();
 
-  vm.get_table_entry("tux");
+  const ssq::Table tux = table.findTable("tux");
   Vector p(0.0f, 0.0f);
-  if (!vm.get_float("x", p.x) || !vm.get_float("y", p.y))
+  if (!tux.get("x", p.x) || !tux.get("y", p.y))
   {
     log_warning << "Player position not set, respawning." << std::endl;
     sector.move_to_spawnpoint(DEFAULT_SPAWNPOINT_NAME);
     m_position_was_reset = true;
   }
-  std::string back_str = vm.read_string("back");
+
+  std::string back_str;
+  tux.get("back", back_str);
   sector.m_tux->m_back_direction = string_to_direction(back_str);
   sector.m_tux->set_tile_pos(p);
 
   int tile_data = sector.tile_data_at(p);
-  if (!( tile_data & ( Tile::WORLDMAP_NORTH | Tile::WORLDMAP_SOUTH | Tile::WORLDMAP_WEST | Tile::WORLDMAP_EAST ))) {
+  if (!(tile_data & (Tile::WORLDMAP_NORTH | Tile::WORLDMAP_SOUTH | Tile::WORLDMAP_WEST | Tile::WORLDMAP_EAST)))
+  {
     log_warning << "Player at illegal position " << p.x << ", " << p.y << " respawning." << std::endl;
     sector.move_to_spawnpoint(DEFAULT_SPAWNPOINT_NAME);
     m_position_was_reset = true;
   }
-  sq_pop(vm.get_vm(), 1);
 }
 
 /** Load levels **/
 void
-WorldMapState::load_levels()
+WorldMapState::load_levels(const ssq::Table& table)
 {
-  WORLDMAP_STATE_SQUIRREL_VM_GUARD;
-  WORLDMAP_STATE_SECTOR_GUARD;
+  try
+  {
+    const ssq::Table levels = table.findTable("levels");
```

Comment:
If you do rename the other one, maybe rename this one to `levels_table` as well.